src/modes/draw_line_string.js

```diff
@@ -1,96 +1,101 @@
 const CommonSelectors = require('../lib/common_selectors');
 const LineString = require('../feature_types/line_string');
 const isEventAtCoordinates = require('../lib/is_event_at_coordinates');
-const doubleClickZoom = require('../lib/double_click_zoom');
 const Constants = require('../constants');
 const createVertex = require('../lib/create_vertex');
 
-module.exports = function(ctx) {
-  const line = new LineString(ctx, {
-    type: Constants.geojsonTypes.FEATURE,
-    properties: {},
-    geometry: {
-      type: Constants.geojsonTypes.LINE_STRING,
-      coordinates: []
+import ModeInterface from './mode_interface';
+
+export default class DrawLineStringMode extends ModeInterface {
+  constructor(store, ui) {
```

Comment:
That is what I'm envisioning, but it's also pretty limiting and I don't want to replicate the full api of store and ui in the mode so I'm not sure yet.